src/test/appdistro/client.spec.ts

```diff
@@ -39,17 +41,44 @@ describe("distribution", () => {
 
   describe("getApp", () => {
     it("should throw error when app does not exist", () => {
-      nock(api.appDistributionOrigin).get(`/v1alpha/apps/${appId}`).reply(404, {});
+      nock(api.appDistributionOrigin)
+        .get(`/v1alpha/apps/${appId}?appView=${appViewBasic}`)
+        .reply(404, {});
       return expect(appDistributionClient.getApp()).to.be.rejected;
```

Comment:
Something that's important with `nock` is to make sure the calls you set up were in fact called. In most cases, the HTTP call would just fail if it didn't match the `nock` setup, but it's a little better to assert that. I'd recommend this pattern throughout:

```typescript
it("should...", async () => {
  nock(things).get(200, "stuff)

  await expect(theThing).to.do.something;

  // This is the only serious line: asserts that there are no more `nock`'d requests.
  expect(nock.isDone()).to.be.true;
})
```

So, if your `nock` didn't get called, the test would fail on the last `expect` (which is good, because it wouldn't have done what you expected).

Could you apply this pattern throughout?

Reply:
Definitely!

Reply:
@bkendall I assume we only care about checking nock.isDone() when we're "nocking". PTAL again at your convenience and see if I missed anything